utils.js

```diff
@@ -61,6 +61,7 @@ exports.scopeMatch = function(scopePatterns, scopesets) {
   assert(scopesets instanceof Array, "scopesets must be a string or an array");
   return scopesets.some(function(scopeset) {
     assert(scopesets instanceof Array, "scopeset must be a string or an array");
+    assert(scopeset.length > 0, "scopeset must have at least one element");
```

Comment:
I'm not sure this is a good idea...

Sometimes the set of scopes you want to check that someone has is computed. And that computation may result in `[]` because we find that no scopes are required... And with this we would have to implement a special case everywhere we compute scope-sets that may be the empty-set.

I'm not sure it's a sign of misconfiguration, it's just a trivial truth. If I write `scopes: []` in the declaration of an API end-point that's probably what I wanted.